docs/content/develop/templates.md

```diff
@@ -0,0 +1,280 @@
+   }
+   ```
+2. Internet Explorer & Edge
+    - There are no slots in IE11 or Edge, so the polyfill **will not**  style slotted elements.
+3. You can only style **direct** descendants of the slot. 
+    - If you need to use nested elements in the web component, like unordered lists & list items: `<ul><li>...</li></ul>` you cannot actually style the list item. This means you would have to ship a light DOM stylesheet with the web component:
+
+
+
+
+### Regular Light DOM stylesheet (i.e. pfe-cta--lightdom.css )
+
+Some web components ship with a light DOM stylesheet for IE / Edge support. These stylesheets are opt-in (they are not included in the JavaScript file for the web component).
+
+*   Includes 
+    *   component fallback styles
+    *   basic class-based styles for typography  
+
+
+
+<br/>
+
+
+## Shadow DOM 
+
+
+### **Pros**
+
+
+
+1. Styles are encapsulated, so there are no worries about conflicting styles or specificity battles from other stylesheets. 
+    - Sometimes this is desirable, like links inside the pfe-cta component. We don't want those links to accept any external styles.
+2. Semantic markup
+    - Would google see an H3 that a dev put in the pfe-hero component, or the h1 that the component upgrades it to?
+
+
+### **Cons**
+
+
+
+1. Analytics tools may not see links in the Shadow DOM. Therefore we must always bubble up custom events if there are interactive items in the shadow DOM.
+2. When content is moved to the shadow DOM on upgrade, it can slow down rendering as the page re-paints
+
+
+
+
+# Understanding web component templates and slots
+
+_**Note:** Elements that can be inserted into slots are known as slotable; when an element has been inserted in a slot, it is said to be slotted._
+
+
+
+*   Slots are places to pass content or markup into specific regions within your web component template.
+
+    ```
+    // my-component.html:
+	<div class="custom">
+	    <slot name="header"></slot>
+	    <slot></slot>
+	    <slot name="footer"></slot>
+	</div>
+    ```
+
+
+*   If you put some content between the opening & closing tags of a web component without a slot name, it will put that markup into the unnamed slot.
+    *   If there is no unnamed slot, it will _not_ render that content. 
+    *   For this reason, it’s a good idea to leave one unnamed slot if general markup is allowed within that component. All PatternFly Element web components operate this way. 
+*   Whenever you add slot="something", you are telling the webcomponent where to put this information within the inner template.
+
+    ```
+    // my-web-page.html:
+	<pfe-card>
+	      <div slot="pfe-card--header">
+	          <h1>This heading text will appear in the card header region</h1>
+	      </div>
+	</pfe-card>
+    ```
+
+
+*   You can’t have nested slots _with the same name_, but when you use different names, the web component will pick up the content and put it where it belongs in the web component template, regardless of order in the light DOM.
```

Comment:
A nuance here, you can use the same slot name in the light dom that the developer is writing.  You just can't reuse a slot name in the template file more than once.

Reply:
On my website:
```
<pfe-demo>
    <h1 slot="foo">Bar</h1>
    <p slot="foo">Testing slots</p>
</pfe-demo>
```

Those would work, they would just render in the same place in the template as siblings.

Reply:
That's true, though it does say "nested slots". Do you have a proposed bit you'd like to add for clarity?

Reply:
Perhaps something like:
---
* Slots should be direct descendants of the component tag - if they are nested inside other markup, the slot will not be picked up by the component as the component is only aware of its own direct children. 
* In the template, slots cannot have the same name however you can put multiple elements inside a slot by assigning that slot multiple times in your light dom code.  Example:
    ```
    <pfe-demo>
        <h1 slot="foo">Bar</h1>
        <p slot="foo">Testing slots</p>
    </pfe-demo>
    ```
    Both the h1 and p tag in this example will render inside the "foo" slot of the component's template.

Reply:
thanks! added / updated